File: divisible sum pairs/Divisible Sum Pairs.c
#include <stdio.h>

int divisibleSumPairs(int n, int k, int* ar) {
    int count = 0;

    for (int i = 0; i < n; i++) {
        for (int j = i + 1; j < n; j++) {
            if ((ar[i] + ar[j]) % k == 0) {
                count++;
            }
        }
    }

    return count;
}

int main() {
    int n, k;
    scanf("%d %d", &n, &k);

    int ar[n];
    for (int i = 0; i < n; i++) {
        scanf("%d", &ar[i]);
    }

    int result = divisibleSumPairs(n, k, ar);
    printf("%d\n", result);

    return 0;
}

File: divisible sum pairs/Divisible Sum Pairs.rs
use std::io::{self, BufRead};

fn divisible_sum_pairs(n: usize, k: i32, ar: &[i32]) -> i32 {
    let mut count = 0;

    for i in 0..n {
        for j in (i + 1)..n {
            if (ar[i] + ar[j]) % k == 0 {
                count += 1;
            }
        }
    }

    count
}

fn main() {
    let stdin = io::stdin();
    let mut input = stdin.lock().lines();

    let nk: Vec<i32> = input
        .next()
        .unwrap()
        .unwrap()
        .split_whitespace()
        .map(|x| x.parse().unwrap())
        .collect();

    let n = nk[0] as usize;
    let k = nk[1];

    let ar: Vec<i32> = input
        .next()
        .unwrap()
        .unwrap()
        .split_whitespace()
        .map(|x| x.parse().unwrap())
        .collect();

    let result = divisible_sum_pairs(n, k, &ar);
    println!("{}", result);
}
